backends/vulkan/runtime/api/containers/Tensor.h

```diff
@@ -21,21 +21,60 @@ namespace api {
 
 static constexpr size_t kTensorDimLimit = 8;
 
+/*
+ * PackedDimInfo encapsulates metadata about packed dimensions in GPU tensors.
+ * This includes information about which dimension is packed, whether it's
+ * padded, and tiled packing information for special layouts like 4W4C and 4H4W.
+ */
+struct PackedDimInfo {
+  // Describes which dimension is "tightly packed" using WHCN index (i.e. 0 for
+  // width, 1 for height, etc.). For texture backed tensors, this describes
+  // which dimension is packed along a texel. For buffer backed tensors, this
+  // describes which dimension has a stride of 1 (i.e. is last in the dim
+  // order).
+  int32_t packed_dim;
+  // Describes if the packed dimension is padded to a multiple of 4. This will
+  // be true for all tensors that use texture storage, and will also be true
+  // for the PACKED_PADDED memory layouts.
+  bool packed_dim_padded;
+  // Describes a second level of packing, if applicable (which will only apply
+  // to the 4W4C and 4H4W layouts). If there is no second level of packing,
+  // then this will be equal to packed_dim. Otherwise, it will represent the
+  // outer dim used to construct tiled packing. For example, 4W4C will have
+  // packed_dim = 2 and outer_packed_dim = 0.
+  int32_t outer_packed_dim;
+  // Whether the outer packed dim is padded to the next multiple of 4. This is
+  // true only for tiled layouts.
+  bool outer_packed_dim_padded;
+
+  PackedDimInfo(
+      int32_t dim,
+      bool dim_padded,
+      int32_t outer_dim,
+      bool outer_dim_padded)
+      : packed_dim(dim),
+        packed_dim_padded(dim_padded),
+        outer_packed_dim(outer_dim),
+        outer_packed_dim_padded(outer_dim_padded) {}
```

Comment:
don't you want to add validation that outer_packed_dim_padded can be only true for tiled layout?